src/main/java/org/isf/visits/gui/VisitView.java

```diff
@@ -296,17 +329,42 @@ private JButton getAddVisitFirstButton() {
 		}
 		return addFirstVisitButton;
 	}
+	
+	private JButton getDeleteFirstVisitButton() {
+		if (deleteFirstVisitButton == null) {
+			deleteFirstVisitButton = new JButton(MessageBundle.getMessage("angal.visit.removevisit.btn"));
+			deleteFirstVisitButton.setMaximumSize(new Dimension(VISIT_BUTTON_WIDTH, ALL_BUTTON_HEIGHT));
+			deleteFirstVisitButton.setHorizontalAlignment(SwingConstants.LEFT);
+			deleteFirstVisitButton.addActionListener(actionEvent -> {
+
+				int row = jTableFirst.getSelectedRow();
+				if (row < 0) {
+					MessageDialog.info(VisitView.this, MessageBundle.getMessage("angal.common.pleaseselectarow.msg"));
+					return;
+				}
+				Visit visit = (Visit) jTableFirst.getModel().getValueAt(row, -1);
+				int ok = MessageDialog.okCancel(VisitView.this, MessageBundle.getMessage("angal.visit.removevisit.msg"));
```

Comment:
The `okCancel()` method does the message lookup for you; only pass the key and remove the `MessageBundle.getMessage()`

Reply:
Done, thanks!